tests/provision_test.go

```diff
@@ -65,6 +65,15 @@ func blockVolCreationTest() {
 	By("verifying ZFSVolume object", VerifyZFSVolume)
 	By("verifying ZFSVolume property change", VerifyZFSVolumePropEdit)
 	By("Deleting application deployment")
+
+	createSnapshot(pvcName, snapName)
+	verifySnapshotCreated(snapName)
+	createClone(clonePvcName, snapName, scObj.Name)
+	By("Creating and deploying clone app pod", createDeployVerifyCloneApp)
+
+	By("Deleting clone and main application deployment")
+	deleteAppDeployment(cloneAppName)
+
 	deleteAppDeployment(appName)
```

Comment:
Please add the cleanup steps as well for the created resources as above in the filesystem one.